sub-packages/bionemo-geneformer/src/bionemo/geneformer/run/main.py

```diff
@@ -28,70 +28,71 @@
+    )
 
-        # Debug options.
-        parser.add_argument(
-            "--nsys-profiling",
-            action="store_true",
-            default=False,
-            help="Enable targeted `nsys` profiling on the training loop for a defined step range. To actually get profiling output you must run the whole program with `nsys`. For example: "
-            " `nsys profile -s none -o output_report_name -t cuda,nvtx --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop  [regular python command here]`",
-        )
-        # start, end, rank
-        parser.add_argument(
-            "--nsys-start-step",
-            type=int,
-            required=False,
-            default=0,
-            help="Start nsys profiling after this step.",
-        )
-        parser.add_argument(
-            "--nsys-end-step",
-            type=int,
-            required=False,
-            help="End nsys profiling after this step.",
-        )
-        # rank as list of integers
-        parser.add_argument(
-            "--nsys-ranks",
-            type=int,
-            nargs="+",
-            required=False,
-            default=[0],
-            help="Enable nsys profiling for these ranks.",
-        )
-        parser.add_argument(
-            "--disable-checkpointing",
-            action="store_false",
-            default=True,
-            dest="create_checkpoint_callback",
-            help="Disable creating a ModelCheckpoint callback.",
-        )
+    # Debug options.
+    parser.add_argument(
+        "--nsys-profiling",
+        action="store_true",
+        default=False,
+        help="Enable targeted `nsys` profiling on the training loop for a defined step range. To actually get profiling output you must run the whole program with `nsys`. For example: "
+        " `nsys profile -s none -o output_report_name -t cuda,nvtx --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop  [regular python command here]`",
+    )
+    # start, end, rank
+    parser.add_argument(
+        "--nsys-start-step",
+        type=int,
+        required=False,
+        default=0,
+        help="Start nsys profiling after this step.",
+    )
+    parser.add_argument(
+        "--nsys-end-step",
+        type=int,
+        required=False,
+        help="End nsys profiling after this step.",
+    )
+    # rank as list of integers
+    parser.add_argument(
+        "--nsys-ranks",
+        type=int,
+        nargs="+",
+        required=False,
+        default=[0],
+        help="Enable nsys profiling for these ranks.",
+    )
+    parser.add_argument(
+        "--disable-checkpointing",
+        action="store_false",
+        default=True,
+        dest="create_checkpoint_callback",
+        help="Disable creating a ModelCheckpoint callback.",
+    )
+
+    return parser.parse_args()
 
-        return parser.parse_args()
 
+def main(args=None):  # noqa: D103
```

Comment:
I think the type of `load_config` should be `str | Type | None` something like that? You call it with the class type as default, which fails to match to the if/else/etc cases so it doesn't get changed.